### ISSUE TITLE
Issue with utils::choose.dir() function in RStudio version 2023.03.0 Build 386 after updating R

### ISSUE THREAD

**Issue by PercyRvx** [rstudio/rstudio#13078]
Dear RStudio Team,

I hope this email finds you well. I am writing to report an issue I have encountered while using RStudio version 2023.03.0 Build 386.

After updating the R engine to version 4.3.0, I have noticed that when I execute the function utils::choose.dir(), it returns NA without opening the folder selection window as expected. However, this issue does not occur when running the function in the R console outside of RStudio.

As a temporary workaround, I have found that if I first invoke the base::file.choose() function and then run choose.dir() within the same RStudio session, the folder selection window does appear.

I wanted to bring this matter to your attention, as it seems to be specific to RStudio and not to the R console itself. I hope that you can look into this issue and provide a resolution or guidance on how to address it.

Thank you for your continued support and for developing an excellent product. I look forward to hearing from you soon.

Best regards,

Percy Rodríguez

**Comment by kevinushey** [comment]
Note for anyone investigating; `choose.files()` and `choose.dir()` are both functions only available on Windows builds of R.

**Comment by timtmok** [comment]
This looks like a problem with Rterm.exe. I don't get the selection dialog.

![image](https://github.com/rstudio/rstudio/assets/9591545/24313d28-ebec-4813-9b4c-3ea8cb6c73e5)

If I use Rgui.exe then I do get the selection dialog.

![image](https://github.com/rstudio/rstudio/assets/9591545/9cc3480e-75f5-44b5-bb8e-4cd924cdccd6)

**Comment by sjackson1997** [comment]
I got here because I'm experiencing the same problem.  Apologize for ignorance, but I don't know what Rterm.exe is (is that what RStudio is running as a console?).  

FWIW, here's what I'm running:
> version
               _                                
platform       x86_64-w64-mingw32               
arch           x86_64                           
os             mingw32                          
crt            ucrt                             
system         x86_64, mingw32                  
status                                          
major          4                                
minor          3.1                              
year           2023                             
month          06                               
day            16                               
svn rev        84548                            
language       R                                
version.string R version 4.3.1 (2023-06-16 ucrt)
nickname       Beagle Scouts           

RStudio 2023.06.1+524 "Mountain Hydrangea" Release (547dcf861cac0253a8abb52c135e44e02ba407a1, 2023-07-07) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.06.1+524 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36

**Comment by timtmok** [comment]
@sjackson1997 Yes, RTerm.exe is how RStudio runs R. Thank you for your report. I don't think there's a workaround for this other than downgrading R.

From my quick investigation, I think RGui works since it is meant to show a graphical interface. RTerm might not support this since it is supposed to be a terminal environment. Maybe the R update changed this behaviour because it aligns with the purpose for RTerm. It's unknown at this time if there's a solution for this.

**Comment by gagecarto** [comment]
Hi All - I've recently run into this issue as upgrading R broke some a Shiny app which depended on the choose.dir() function. Any suggestion on working around this? From what I am reading it sounds like this is a problem with R and thus R needs to be downgraded to fix the problem. 

Many thanks

**Comment by ismirsehregal** [comment]
I also just came across this problem - same behaviour when using Rscript.exe, but I can't seem to find any "official" news about this.

@gagecarto regarding shiny you might want to check:
https://github.com/thomasp85/shinyFiles

**Comment by jeroen** [comment]
This might be a dll conflict with system dll's, see: https://github.com/r-lib/textshaping/issues/36#issuecomment-2578053982

**Comment by saulo1305** [comment]
Any fixes or workarounds on this issue? It`s been 3 years I'm getting this error. The issue was first reported at R 4.3. We are at R 4.6 and still no fix.

**Comment by torden81** [comment]
> Any fixes or workarounds on this issue? It`s been 3 years I'm getting this error. The issue was first reported at R 4.3. We are at R 4.6 and still no fix.

Hi,
as a workaround, using `tcltk::tk_choose.dir()` works for me.

Apart from that, you might have more success getting this issue addressed by reporting it to the R project directly, as it seems to be unrelated to RStudio.
For issues in base packages like {utils}, `bug.report(package = "utils")` points to: https://bugs.r-project.org/

**Comment by kevinushey** [comment]
Should (finally) be resolved, with a fix to be included with the upcoming release in August.
